recipes-kernel/linux/linux-dm-crypt.inc

```diff
@@ -1,7 +1,4 @@
-SRC_URI += " \
+SRC_URI:append = "\
     file://dm-crypt.cfg \
-"
-
-SRC_URI:append:imx-generic-bsp = " \
-    file://caam.cfg \
+    file://${TDX_ENC_KEY_BACKEND}.cfg \
```

Comment:
Out of curiosity, what would happen if one tries to enable CAAM on a non-NXP device? I'd guess there would be errors during kernel build in such case my question would be: do we want to prevent them? (I'm ok either way)

Reply:
The build will not fail. The kernel will be compiled with CAAM support (which is supported upstream), even on non-iMX devices.

The error will be at runtime. In case there is no CAAM device available, the Trusted Keys subsystem will fail to create a key backed by CAAM.

Reply:
Nice. So the user should be able to realize they did something wrong.